packages/react-utilities/src/hooks/useEvent.ts

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+
+type UseEvent = {
+  /**
+   * The element to listen to.
+   */
+  element?: Element | Window | Document;
+
+  /**
+   * The event type to watch.
+   */
+  type: string;
+
+  /**
+   * The callback to call when the eventName is triggered.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  callback: (ev: any) => void;
```

Comment:
It might be worth adding a generic (with a default of `Event`) for the event type.